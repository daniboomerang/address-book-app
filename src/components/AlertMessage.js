import React from 'react';

/**
 * Alert message displayed at the bottom of the page
 */
const AlertMessage = ({ children }) => (
  <div
    data-testid="alert-message"
    className="z-10 border-t-4 bg-primary border-red-600 text-center py-2 lg:px-4 fixed bottom-0 left-0 w-full"
  >
    <div className="font-semibold text-red-600 px-4 text-center flex-auto">{children}</div>
  </div>
);

export default AlertMessage;
