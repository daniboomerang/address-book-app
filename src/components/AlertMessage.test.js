import React from 'react';
import { mount } from 'enzyme';
import AlertMessage from './AlertMessage';

describe('AlertMessage component', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <AlertMessage>
        <div data-testid="some-children-id">Alert!</div>
      </AlertMessage>
    );

    expect(wrapper.find(`div[data-testid="alert-message"]`).length).toBe(1);
    expect(wrapper.find(`div[data-testid="some-children-id"]`).length).toBe(1);
    expect(wrapper.find(`div[data-testid="some-children-id"]`).text()).toBe('Alert!');
    expect(wrapper).toMatchSnapshot();
  });
});
