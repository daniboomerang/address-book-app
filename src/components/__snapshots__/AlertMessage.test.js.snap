// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage component should render correctly 1`] = `
<AlertMessage>
  <div
    className="z-10 border-t-4 bg-primary border-red-600 text-center py-2 lg:px-4 fixed bottom-0 left-0 w-full"
    data-testid="alert-message"
  >
    <div
      className="font-semibold text-red-600 px-4 text-center flex-auto"
    >
      <div
        data-testid="some-children-id"
      >
        Alert!
      </div>
    </div>
  </div>
</AlertMessage>
`;
